### PR TITLE
Implement the update api on fs.

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

As titled.

Related issue number
--------------------

Fixes part of https://github.com/v6d-io/v6d/issues/1841

### CODE REVIEW

File: modules/llm-cache/storage/file_storage.cc [hunk 1]
```diff
@@ -39,78 +41,48 @@ namespace vineyard {
...
+  std::vector<std::string> pathList;
+  std::string dir = rootPath;
+  std::string path;
+  int tokenSize;
+
+  VINEYARD_ASSERT(batchSize > 0);
+  // check if the root path ends with '/'
+  if (dir.back() != '/') {
+    dir += "/";
+  }
```
Comment (vegetableysm) [inline modules/llm-cache/storage/file_storage.cc:53]: Use std::regex_replace(path, std::regex("/+"), "/") to deal with the final path may be better.

File: modules/llm-cache/storage/file_storage.cc [hunk 2]
```diff
@@ -39,78 +41,48 @@ namespace vineyard {
...
-        for (int k = 0; k < kvStateSize; k++) {
-          data += std::to_string((
-                      reinterpret_cast<uint8_t*>(kvStateList[i * batchSize + j]
-                                                     .find(currentLayer)
-                                                     ->second.first.data))[k]) +
-                  " ";
+    std::filesystem::path filePath(dir + pathList[i]);
+    RETURN_ON_ERROR(Mkdir(filePath.parent_path().string()));
+    RETURN_ON_ERROR(Open(filePath.string(), fd, FileOperationType::WRITE));
+    RETURN_ON_ERROR(Seek(fd, 0));
```
Comment (vegetableysm) [inline modules/llm-cache/storage/file_storage.cc:66]: Currently, when open a file, the read/write pointer is at the begin of the file.

File: modules/llm-cache/storage/file_storage.cc (hunk 1 above)
Comment (vegetableysm) [reply modules/llm-cache/storage/file_storage.cc:53]: e.g.  user the path as rootPath + "/" + pathList[i]
Then, use the std::regex_replace to replace the multi "/" to "/"

**Review by vegetableysm — APPROVED** [review]

File: modules/llm-cache/storage/file_storage.cc (hunk 2 above)
Comment (dashanji) [reply modules/llm-cache/storage/file_storage.cc:66]: Make sense to me. Thanks

File: modules/llm-cache/storage/file_storage.cc (hunk 1 above)
Comment (dashanji) [reply modules/llm-cache/storage/file_storage.cc]: Based on our discussion, the filesystem api will do the regex itself. E,g, `/vineyard/llm/cache` == `/vineyard////llm//cache`

**Pull request merged**